kong/plugins/hmac-auth/access.lua

```diff
@@ -81,10 +81,24 @@ local function create_hash(request, hmac_params, headers)
   return ngx_sha1(hmac_params.secret, signing_string)
 end
 
+local function is_digest_equal(digest_1, digest_2)
+  if #digest_1 ~= #digest_1 then
+    return false
+  end
+      
+  local result = true
+  for i=1, #digest_1 do
+    if digest_1:sub(i, i) ~= digest_2:sub(i, i) then
+      result = false
```

Comment:
I suggest adding a `break` after this. Not point on continuing after a difference was found.